platform/database/tests/database_test.go

```diff
@@ -810,4 +810,140 @@ func TestProdDatabase(t *testing.T) {
+			}
+
+			if data.Code != expectedData.Code {
+				return fmt.Errorf("expected to retrieve the second data, but the code does not match. Expected: %s, received: %ss", expectedData.Code, data.Code)
+			}
+
+			return nil
+		}
+
+		runSelectTransactionTest := func(db database.ReadDB, expectedData testData) error {
+			ctx := context.Background()
+			tx, err := db.Begin()
+			if err != nil {
+				return fmt.Errorf("can't start transaction: %v", err)
+			}
+
+			var data []testData
+			if err := tx.SelectContext(ctx, &data, `SELECT * FROM test_data WHERE id = $1`, expectedData.ID); err != nil {
+				return fmt.Errorf("can't execute statement: %v", err)
+			}
+
+			if err := tx.Commit(); err != nil {
+				return fmt.Errorf("can't commit the transaction: %v", err)
+			}
+
+			if len(data) != 1 {
+				return fmt.Errorf("expected to retrieve only 1 item: %v", err)
+			}
+
+			if data[0].Code != expectedData.Code {
+				return fmt.Errorf("expected the to retrieve the first data element, but the code does not match. Expected: %s, received: %s", expectedData.Code, data[0].Code)
+			}
+
+			return nil
+		}
+
+		writeDatabase, err := database.ConnectWriteDatabase(cfg)
+		if err != nil {
+			t.Fatalf("could not establish connection to write DB: %#v", err)
+		}
+
+		if err := runInsertTest(writeDatabase, firstData); err != nil {
+			t.Fatalf("The insert command failed: %v", err)
+		}
+
+		if err := runInsertTransactionTest(writeDatabase, secondData); err != nil {
+			t.Fatalf("The transaction command failed: %v", err)
+		}
+
+		readDatabase, err := database.ConnectReadDatabase(cfg)
+		if err != nil {
+			t.Fatalf("could not establish connection to write DB: %#v", err)
+		}
+
+		if err := runSelectTest(readDatabase, secondData); err != nil {
+			t.Fatalf("The select command failed: %v", err)
+		}
+
+		if err := runSelectTransactionTest(readDatabase, firstData); err != nil {
+			t.Fatalf("The transaction command failed: %v", err)
+		}
+
+		sandBoxReadDB, sandboxWriteDB, err := database.SandboxReadWriteConnect(cfg)
+		if err != nil {
+			t.Fatalf("could not connect to the sandbox DB: %v", err)
+		}
+		defer func() {
+			sandBoxReadDB.Close()
+			sandboxWriteDB.Close()
+		}()
+
+		thirdData := testData{
+			ID:   "ee320876-5b21-416b-bf4c-8c8a5dfd4727",
+			Code: "third_data",
+		}
+
+		fourthData := testData{
+			ID:   "d8ac2183-c167-429c-b7b0-020644c40f5f",
+			Code: "fourth_data",
+		}
+
+		if err := runInsertTest(sandboxWriteDB, thirdData); err != nil {
+			t.Fatalf("The insert command failed: %v", err)
+		}
+
+		if err := runInsertTransactionTest(sandboxWriteDB, fourthData); err != nil {
+			t.Fatalf("The transaction command failed: %v", err)
+		}
+
+		if err := runSelectTest(sandBoxReadDB, thirdData); err != nil {
+			t.Fatalf("The select command failed: %v", err)
+		}
+
+		if err := runSelectTransactionTest(sandBoxReadDB, fourthData); err != nil {
+			t.Fatalf("The transaction command failed: %v", err)
+		}
+	})
 }
```

Comment:
wouldn't this be better in `sandbox_test.go`  ?

Reply:
you are right